--- vectorscan-vectorscan-5.4.12/src/fdr/fdr_engine_description.cpp~	2026-05-08 15:13:31.000000000 +0200
+++ vectorscan-vectorscan-5.4.12/src/fdr/fdr_engine_description.cpp	2026-05-08 15:19:39.180015099 +0200
@@ -71,7 +71,7 @@
         } else if (num_lits < 5000) {
             // for larger but not huge sizes, go to stride 2 only if we have at
             // least minlen 3
-            desiredStride = std::min(min_len - 1, 2UL);
+            desiredStride = std::min(min_len - 1, static_cast<size_t>(2));
         }
     }
 
